libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Store/SFSDKEventStoreManager.m

```diff
@@ -96,15 +96,27 @@ - (void) storeEvent:(SFSDKInstrumentationEvent *) event {
     if (encryptedData) {
         NSString *filename = [self filenameForEvent:eventCopy.eventId];
         NSString *parentDir = [filename stringByDeletingLastPathComponent];
-        [[NSFileManager defaultManager] createDirectoryAtPath:parentDir withIntermediateDirectories:YES attributes: @{ NSFileProtectionKey: NSFileProtectionCompleteUntilFirstUserAuthentication } error:&error];
-        [encryptedData writeToFile:filename options:NSDataWritingFileProtectionCompleteUntilFirstUserAuthentication error:&error];
+        NSFileManager *fileManager = [NSFileManager defaultManager];        
+        [fileManager createDirectoryAtPath:parentDir withIntermediateDirectories:YES attributes: @{ NSFileProtectionKey: NSFileProtectionCompleteUntilFirstUserAuthentication } error:&error];
+        
         if (error) {
-            [SFSDKAnalyticsLogger w:[self class] format:@"Error occurred while writing to file: %@", error.localizedDescription];
-        } else {
-            @synchronized (self.eventCountMutex) {
-                self.numStoredEvents++;
+            [SFSDKAnalyticsLogger w:[self class] format:@"Error occurred while trying to create directory: %@", error.localizedDescription];
+            return;
+        }
+        
+        @try {
```

Comment:
Other than restructuring the code, it looks like the only change here is adding a `try-catch` block. Do we know what exception is thrown, and under what circumstances this happens? Are we able to reproduce this issue? I'm wondering if we could address the root cause without adding a `try-catch`.

Reply:
@bhariharan I did some more digging, and I can reproduce the exact same stack trace with a crash if the `fileName` property is nil. Digging a little more, it looks like the file name is constructed from a constructor property that comes from the `SFSDKSalesforceAnalyticsManager` init code:
```
NSString *rootStoreDir;
if (_userAccount != nil) {
    rootStoreDir = [[SFDirectoryManager sharedManager] directoryForUser:userAccount type:NSDocumentDirectory components:@[ kEventStoresDirectory ]];
} else {
    rootStoreDir = [[SFDirectoryManager sharedManager] globalDirectoryOfType:NSDocumentDirectory components:@[ kEventStoresDirectory ]];
}
```

It looks like the directory manager has several paths that could return nil in either of these two cases. I have also confirmed that the code changes in this PR prevent the crash. I'm assuming the side effects are that analytics events are not recorded. I would probably suggest someone with more domain knowledge of these singleton classes take a look as to why the directory manager returns a nil store directory.

Reply:
@mbwimmer Thank you for the analysis. I understand that this could occur if `filename` is `nil`, but looking at that code, `filename` should never be `nil`. Are you able to repro a situation where `filename` is `nil`? In other words, `filename` could only be `nil` is the `userAccount` object is not `nil` but it's corrupted inside and has `nil` properties, and this isn't valid. So, I'm wondering how this ever happens.